### PR TITLE
Tag repository

### PR DESCRIPTION
**Pull request by EkaterinaKaonashi**

### CODE REVIEW

File: DevEdu.API/DevEdu.API/Controllers/CourseController.cs [hunk 1]
```diff
@@ -4,15 +4,25 @@
...
+using DevEdu.DAL.Models;
 
 namespace DevEdu.API.Controllers
 {
     [ApiController]
     [Route("api/[controller]")]
     public class CourseController : Controller
     {
-        public CourseController()
+        CourseRepository _courseRepository;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/CourseController.cs:17]: please register these classes as scoped services in DI-container (via interface) and inject them in constructor

File: DevEdu.API/DevEdu.API/Controllers/MaterialController.cs [hunk 2]
```diff
@@ -11,7 +12,11 @@ namespace DevEdu.API.Controllers
     [Route("api/[controller]")]
     public class MaterialController : Controller
     {
-        public MaterialController() { }
+        TagRepository _tagRepository;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/MaterialController.cs:15]: please register this class as a scoped service in DI-container (via interface) and inject it in constructor

File: DevEdu.API/DevEdu.API/Controllers/MaterialController.cs [hunk 3]
```diff
@@ -52,13 +57,15 @@ public string DeleteMaterial(int id)
         [HttpPost("{materialId}/tag/{tagId}")]
         public int AddTagToMaterial(int materialId, int tagId)
         {
+            _tagRepository.InsertTagToTagMaterial(materialId, tagId);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/MaterialController.cs:60]: this method should be renamed to AddTagToMaterial, right?

File: DevEdu.API/DevEdu.API/Controllers/MaterialController.cs [hunk 4]
```diff
@@ -52,13 +57,15 @@ public string DeleteMaterial(int id)
...
             return 1;
         }
 
         // api/material/{materialId}/tag/{tagId}
         [HttpDelete("{materialId}/tag/{tagId}")]
         public string DeleteTagFromMaterial(int materialId, int tagId)
         {
+            _tagRepository.DeleteTagFromTagMaterial(materialId, tagId);
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.API/Controllers/MaterialController.cs:68]: and "DeleteTagFromMaterial" here

File: DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+namespace DevEdu.DAL.Repositories
+{
+    public abstract class BaseRepository
+    {
+        protected const string ConnectionString =
+            @"Data Source=80.78.240.16;Initial Catalog = DevEdu; Persist Security Info=True;User ID = student;Password=qwe!23;";
+
+        protected IDbConnection _connection;
+        protected string _insertProcedure;
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs:12]: please remove this strings from this class

File: DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+using Dapper;
+using DevEdu.DAL.Models;
+
+namespace DevEdu.DAL.Repositories
+{
+    public class CourseRepository : BaseRepository, ICourseRepository
+    {
+        public CourseRepository()
+        {
+            _insertProcedure = "dbo.Course_Insert";
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/CourseRepository.cs:13]: please create private consts for these strings in the top of the class

File: DevEdu.API/DevEdu.DAL/Repositories/TagRepository.cs [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+using System.Data.SqlClient;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Dapper;
+
+
+namespace DevEdu.DAL.Repositories
+{
+    public class TagRepository : BaseRepository
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/TagRepository.cs:13]: please fix methods' names in this class

File: DevEdu.API/DevEdu.DAL/Repositories/TagRepository.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        public int InsertTagToTagMaterial(int materialId, int tagId)
+        {
+            return _connection
+                .QuerySingle(
+                "[dbo].[Tag_Material_Insert]",
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/TagRepository.cs:19]: and use private consts for the procedures' names

File: DevEdu.API/DevEdu.DAL/Repositories/TopicRepository.cs [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+using DevEdu.DAL.Models;
+
+namespace DevEdu.DAL.Repositories
+{
+    public class TopicRepository : BaseRepository
+    {
+        public int AddTopicToCourse(CourseTopicDto dto)
+        {
+            return _connection.QuerySingle<int>(
+                "[dbo].[Course_Topic_Insert]",
```
Comment (Negodyaj) [inline DevEdu.API/DevEdu.DAL/Repositories/TopicRepository.cs:17]: and here

File: DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs (hunk 5 above)
Comment (EkaterinaKaonashi) [reply DevEdu.API/DevEdu.DAL/Repositories/BaseRepository.cs:12]: From 9 until 12 ? 
image

**Pull request merged**